src/PythNetworkHTTPClient.ts

```diff
@@ -0,0 +1,153 @@
+import { AccountInfo, Cluster, clusterApiUrl, Commitment, Connection, PublicKey } from "@solana/web3.js";
+import { Product, PriceData, getPythProgramKeyForCluster, parseProductData, parsePriceData, parseBaseData, AccountType } from ".";
+
+const ONES = '1111111111111111111111111111111';
+
+interface PriceRawData {
+    key: PublicKey;
+    account: AccountInfo<Buffer>;
+}
+
+/**
+ * Reads Pyth price data from a solana web3 connection. This class uses a single HTTP call.
+ * Use the method refreshData() to update prices values.
+ */
+export class PythNetworkHTTPClient {
+    private _program_key: PublicKey;
+    private _commitment: Commitment;
+    private _cluster: Cluster;
+    private _price_queue: PriceRawData[];
+    private productAccountKeyToProduct: Record<string, Product>;
+    private _asset_types: Set<string>;
+    private _product_symbols: Set<string>;
+    private _products: Set<Product>;
+    private _product_price: Map<Product, PriceData>;
+    public priceAccountKeyToProductAccountKey: Record<string, string>;
+
+    constructor(cluster_name: Cluster, commitment: Commitment = 'finalized') {
```

Comment:
can you make this constructor have the same signature as PythConnection -- specifically, take a Connection instead of the name of the cluster? We chose that signature intentionally so that users can configure their RPC node.